### ISSUE TITLE
Extraction Cost Example Incorrect Now

### ISSUE THREAD

**Issue by saulshanabrook** [egraphs-good/egglog#128]
I believe the extraction cost example is no longer accurate:

```
(datatype Expr
  (Num i64 :cost 5))

(define x (Num 1) :cost 10)
(define y (Num 2) :cost 1)

(extract x) ;; (Num 1)
(extract y) ;; (y)
```

Running it yields:

```
./target/debug/egg-smol  tests/extraction-cost.egg 
[INFO ] Declared sort Expr.
[INFO ] Declared function Num.
[INFO ] Run (let v1___ 1).
[INFO ] Run (let v0___ (Num v1___)).
[INFO ] Run (let x v0___).
[INFO ] Run (let v3___ 2).
[INFO ] Run (let v2___ (Num v3___)).
[INFO ] Run (let y v2___).
[INFO ] Run (let v4___ x).
[INFO ] Extracted with cost 6: (Num 1)
[INFO ] Run (let v5___ y).
[INFO ] Extracted with cost 6: (Num 2)
```

I think some of the changes to how define works may have broken this example. I am not sure if this was intentional or not.

**Comment by yihozhang** [comment]
Oh yeah `cost` is [ignored](https://github.com/mwillsey/egg-smol/blob/main/src/ast/desugar.rs#L385) during the recently added desugaring phase. Maybe @oflatt has thoughts on how to add it back?

**Comment by oflatt** [comment]
Yeah, we should deprecate define (I think they are the same now)

**Comment by yihozhang** [comment]
This issue could potentially be closed by #141 once it's ready.

**Comment by mwillsey** [comment]
@yihozhang if that's true then please edit that PRs description to include the string "Closes #141"

**Comment by yihozhang** [comment]
Fixed by #176.
